jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt

```diff
@@ -108,20 +107,27 @@ class ValidatingPanel internal constructor(
         validationAlarm.addRequest(validateRequest, VALIDATION_INTERVAL_MS, ModalityState.stateForComponent(this))
     }
 
-    private inner class ValidatingAction(private val listener: (ActionEvent) -> Unit) : AbstractAction() {
-        override fun actionPerformed(e: ActionEvent) {
-            val errorList = performValidation()
-            if (errorList.isNotEmpty()) {
-                // Give the first error focus
-                val info = errorList.first()
-                info.component?.let {
-                    IdeFocusManager.getInstance(null).requestFocus(it, true)
-                }
+    fun hasValidInputs(): Boolean {
```

Comment:
method name doesn't convey that it has side effects, which is weird imo

Reply:
I can agree with that.

What do you think about `doValidation`?

Reply:
run is fine